components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -239,9 +265,40 @@ private <T> T deserialize(HttpEntity responseEntity, Class<T> returnType) throws
         }
     }
 
+    private ActionInvocationResponse.APIResponse deserialize(HttpEntity responseEntity)
+            throws ActionInvocationException {
+
+        if (!isAcceptablePayload(responseEntity)) {
+            throw new ActionInvocationException("The response content type is not application/json.");
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            String jsonResponse = EntityUtils.toString(responseEntity);
+            JsonNode rootNode = objectMapper.readTree(jsonResponse);
+            boolean isSuccess = ActionExecutionApiClientConstants.successStatus.equalsIgnoreCase(
+                    rootNode.path(ActionExecutionApiClientConstants.actionStatus).asText());
+            if (isSuccess) {
+                return objectMapper.readValue(jsonResponse, ActionInvocationSuccessResponse.class);
+            } else {
+                return objectMapper.readValue(jsonResponse, ActionInvocationFailureResponse.class);
+            }
+        } catch (IOException e) {
+            throw new ActionInvocationException("Parsing JSON response failed.", e);
+        }
+    }
+
     private boolean isAcceptablePayload(HttpEntity responseEntity) {
 
         return responseEntity != null && responseEntity.getContentType() != null &&
                 responseEntity.getContentType().getValue().contains("application/json");
     }
+
+    private static class ActionExecutionApiClientConstants {
```

Comment:
Is there any reason why we have a separate private class for constants instead define them as just constants in the main class.